packages/devtools-reps/src/object-inspector/tests/component/window.js

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const { mount } = require("enzyme");
+const React = require("react");
+const { createFactory } = React;
+const ObjectInspector = createFactory(require("../../index"));
+const {
+  createNode,
+} = require("../../utils");
+
+const gripWindowStubs = require("../../../reps/stubs/window");
+
+describe("ObjectInspector - dimTopLevelWindow", () => {
+  it("renders collapsed top-level window when dimTopLevelWindow is true", () => {
+    const windowNode = createNode(
+      null,
+      "window",
+      "windowpath",
+      {value: gripWindowStubs.get("Window")}
+    );
+
+    // The window node should have the "lessen" class when collapsed.
+    let oi = mount(ObjectInspector({
+      autoExpandDepth: 0,
+      roots: [windowNode],
+      dimTopLevelWindow: true,
+      getObjectProperties: () => {},
+      loadObjectProperties: () => {},
+    }));
+    expect(oi.find(".node.lessen").exists()).toBeTruthy();
+  });
+
+  it("renders expanded top-level window when dimTopLevelWindow is true", () => {
+    const windowNode = createNode(
+      null,
+      "window",
+      "windowpath",
+      {value: gripWindowStubs.get("Window")}
+    );
+
+    // The window node should not have the "lessen" class when expanded.
+    const oi = mount(ObjectInspector({
+      autoExpandDepth: 1,
+      roots: [windowNode],
+      dimTopLevelWindow: true,
+      getObjectProperties: () => {},
+      loadObjectProperties: () => {},
+    }));
+    expect(oi.find(".node.lessen").exists()).toBeFalsy();
```

Comment:
this could also benefit from snapshot tests:

```js
    expect(oi.find(".node.lessen").exists()).toBeFalsy();
    expect(oi).toMatchSnasphot()
```

We use both tests in a lot of tests to articulate the specific requirements while also guarding agains unexpected changes. It's nice to see when something small changes like a tag name goes from a span to an li and the snapshot picks it up. Most of the time, it doesn't matter but it does queue you in to surprises.